File: server/user-service/src/main/java/com/mshop/userservice/service/UserService.java
package com.mshop.userservice.service;

import com.sudo248.domain.exception.ApiException;
import com.mshop.userservice.controller.dto.UserDto;
import com.mshop.userservice.controller.dto.UserInfoDto;
import com.mshop.userservice.repository.entitity.User;

import java.util.List;

public interface UserService {
    UserDto getUser(String userId);

    List<UserInfoDto> getAllUserInfo() throws ApiException;
    UserInfoDto getUserInfo(String userId) throws ApiException;
    UserDto postUser(UserDto userDto) throws ApiException;
    UserDto putUser(String userId, UserDto userDto) throws ApiException;
    UserDto toDto(User user);
    User toEntity(UserDto userDto);
}

File: server/product-service/src/main/kotlin/com/mshop/productservice/service/UserService.kt
package com.mshop.productservice.service

import com.mshop.productservice.dto.AddressDto
import com.mshop.productservice.dto.UserInfoDto

interface UserService {

    suspend fun getAllCustomer():List<UserInfoDto>
    suspend fun getUserInfo(userId: String): UserInfoDto

    suspend fun postAddress(address: AddressDto): AddressDto

    suspend fun getAddressById(addressId: String): AddressDto
}